nx/test/nx/defn_test.exs

```diff
@@ -1319,6 +1319,36 @@ defmodule Nx.DefnTest do
       end
     end
 
+    test "undefined local function" do
+      assert_raise CompileError,
+                   ~r"#{location(+5)}: undefined function do_add/2 \(there is no such import\)",
+                   fn ->
+                     defmodule Sample do
+                       import Nx.Defn
+
+                       defn add(a, b) do
+                         do_add(a, b)
+                       end
+                     end
+                   end
+    end
+
+    test "non-defn local function" do
+      assert_raise CompileError,
+                   ~r"#{location(+5)}: cannot use function do_add/2 inside defn because it was not defined with defn",
+                   fn ->
+                     defmodule Sample do
+                       import Nx.Defn
+
+                       defn add(a, b) do
+                         do_add(a, b)
+                       end
+
+                       defp do_add(a, b), do: a + b
+                     end
```

Comment:
@jonatanklosko perhaps this needs to be a string called with Code.eval_string or something like that for tests to pass?

Reply:
Ohh, I see the other two tests fail now as I run the formatter!

Reply:
We assert line relatively and formatting just moved it around. I've just fixed it. The remaining error is what I mentioned above.